Formula/m/mariadb@11.4.rb

```diff
@@ -0,0 +1,186 @@
+class MariadbAT114 < Formula
+  desc "Drop-in replacement for MySQL"
+  homepage "https://mariadb.org/"
+  # TODO: Build with `-DWITH_LIBFMT=system` when fmt >= 11
+  url "https://archive.mariadb.org/mariadb-11.4.3/source/mariadb-11.4.3.tar.gz"
+  sha256 "6f0017b9901bb1897de0eed21caef9ffa9d66ef559345a0d8a6f011308413ece"
+  license "GPL-2.0-only"
+
+  livecheck do
+    url "https://downloads.mariadb.org/rest-api/mariadb/all-releases/?olderReleases=false"
+    strategy :json do |json|
+      json["releases"]&.map do |release|
+        next unless release["release_number"]&.start_with?(version.major_minor)
+        next if release["status"] != "stable"
+
+        release["release_number"]
+      end
+    end
+  end
+
+  keg_only :versioned_formula
+
+  # See: https://mariadb.com/kb/en/changes-improvements-in-mariadb-11-4/
+  # End-of-life on 2029-05-29: https://mariadb.org/about/#maintenance-policy
+  deprecate! date: "2029-05-29", because: :unsupported
```

Comment:
I went with `deprecate!` instead of `disable!` as current `brew` message output is awkward, e.g. with `disable!`:
```
Deprecated because it is not supported upstream! It will be disabled on 2029-05-29.
```

That message would be incorrect as it is supported upstream until 2029-05-29. I think we used to output a better message (maybe only considering a formula "deprecated" a couple months before disable).